ocaml/testsuite/tests/typing-local/exclave.ml

```diff
@@ -214,6 +214,27 @@ Error: This function or one of its parameters escape their region
        when it is partially applied.
 |}]
 
+let f () =
+  exclave_ (
+    (fun x -> function | "a" -> () | _ -> ()) : (string -> string -> unit)
+  )
+[%%expect{|
+Line 3, characters 4-45:
+3 |     (fun x -> function | "a" -> () | _ -> ()) : (string -> string -> unit)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: This function or one of its parameters escape their region
+       when it is partially applied.
+|}]
+
+(* For n-ary functions, inner functions are not constrained *)
```

Comment:
```suggestion
(* For nested functions, inner functions are not constrained *)
```

Reply:
I personally think "n-ary" is slightly better than "nested", as it points to a specific thing in our system, whereas "nested" could be vague: `fun a b -> ...` is still a nested function to some people.